_posts/2025-08-26-strimzi-metrics-reporter.md

```diff
@@ -0,0 +1,97 @@
+---
+layout: post
+title: "Using Strimzi Metrics Reporter to Expose Kafka Metrics in Prometheus Format"
+date: 2025-10-01
+author: Owen Corrigan
+---
+When deploying new Kafka cluster it may be one of your intentions to monitor the cluster using metrics.
+`Apache Kafka®` generates and exposes metrics that reflect how a system is operating. 
+These metrics are useful for monitoring, troubleshooting, tuning, and capacity planning when it comes to running your Kafka cluster.
+In essence, monitoring is crucial to ensure the health and performance of your Kafka clusters.
+`Prometheus` as a metrics monitoring solution has become increasingly common due to its popularity in the cloud-native ecosystem, and it is this format that we use to expose metrics.
+
+The [Strimzi Metrics Reporter](https://github.com/strimzi/metrics-reporter) project implements the Java `MetricsReporter` interface to expose metrics via `Prometheus` and it is available since Strimzi 0.48.0 release.
+
+Up to this point, we have recommended using the `JMX Exporter` as a way to expose Kafka metrics, and we have provided some example yaml files in our [Strimzi Examples folder](https://github.com/strimzi/strimzi-kafka-operator/tree/main/packaging/examples/metrics).
+However, using the JMX Exporter adds unnecessary complexity.
+However, using the JMX Exporter introduces unnecessary complexity. 
+Metrics are first exposed by Kafka through JMX, and then a Java agent scrapes those JMX MBeans, applies mapping rules, and re-exposes them over HTTP for Prometheus. 
+This extra indirection adds several drawbacks:
+* Overhead inside the JVM – The Java agent continuously scrapes and transforms metrics, consuming additional CPU and memory.
```

Comment:
Not sure I would describe it as "continuously scrapes", I think this only happens when the HTTP metrics endpoint is queried. The main issue is that filtering happens when the endpoint is hit, while with the reporter the filtering happens when metrics are registered.